source/AMDSTCPDataServer.cpp

```diff
@@ -365,8 +346,31 @@ void AMDSTCPDataServer::onClientSentRequest(const QString &clientKey)
 			onClientRequestProcessed(clientRequest);
 		}
 	}
-	else
+	else {
+		if(clientRequest->requestType() == AMDSClientRequestDefinitions::Continuous){
+			AMDSClientContinuousDataRequest *clientContinuousDataRequest = qobject_cast<AMDSClientContinuousDataRequest*>(clientRequest);
+			if(clientContinuousDataRequest){
+				if (clientContinuousDataRequest->isHandShakingMessage()) {
+					QString socketKey = clientContinuousDataRequest->handShakingSocketKey();
+					AMDSClientContinuousDataRequest *handShakingclientDataRequest = qobject_cast<AMDSClientContinuousDataRequest*>(continuousDataRequests_.value(socketKey));
+					if (handShakingclientDataRequest) {
+						handShakingclientDataRequest->setHandShakeTime(QDateTime::currentDateTime());
+					} else {
+						AMDSErrorMon::alert(this, 0, QString("Didn't find hand shaking message with socketKey (%1)").arg(socketKey));
+					}
+
+					clientContinuousDataRequest->deleteLater();
+					requestingSocket->disconnectFromHost();
+					return;
+				} else {
+					continuousDataRequests_.insert(clientContinuousDataRequest->socketKey(), clientContinuousDataRequest);
```

Comment:
If this is handled somewhere else, I apologize.  What removes requests from this list/map?